File: javascript-programming/examples/intro/map.js
"use strict";

var numbers = [2, 5, 3, 7];
function double(n) {
    return n*2;
}

var doubles = numbers.map(double);
console.log(numbers);  // [2,  5, 3,  7]
console.log(doubles);  // [4, 10, 6, 14]

var triples = numbers.map(function(n) { return n * 3 });
console.log(triples);  // [6, 15, 9, 21]

File: dart/examples/dart-intro/map.dart
main() {
  List numbers = [1, 2, 3];
  print(numbers);      // [1, 2, 3]
  
  var doubles = numbers.map((val) => 2 * val);
  print(doubles);  // (2, 4, 6)
  
  
  var double_list = doubles.toList();
  print(double_list);  // [2, 4, 6]
}
